PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/referencebook/AzureSdkArtifactGroupPanel.java

```diff
@@ -58,7 +66,14 @@ private void clear() {
     }
 
     private void onPackageOrVersionSelected(AzureSdkArtifactEntity pkg, String version) {
-        this.viewer.setText(pkg.generateMavenDependencySnippet(version));
+        this.pkg = pkg;
+        this.version = version;
+        this.viewer.setText(pkg.generateDependencySnippet(type, version));
+    }
+
+    private void onDependencyTypeSelected(String type) {
+        this.type = type;
+        this.viewer.setText(pkg.generateDependencySnippet(type, version));
```

Comment:
Do we also need to change the file type for text editor? Could get it by `FileTypeManagerEx.getInstance().getFileTypeByExtension("gradle")`